Add new review form

Kage-js opened the pull request:
Created an "add new review" form in react per Casi's notes. Made a review api endpoint.

Kage-js committed:
moved show page to stops/show, added review form 

~unfinished~

Kage-js committed:
converted new review form to react. added api test. delete whitespace.

Kage-js committed:
adding foundation media objects for styling

Kage-js committed:
delete pry

Kage-js committed:
Merge branch 'master' into add_new_review_form

CassandraNewell commented on app/controllers/api/v1/reviews_controller.rb:
```diff
@@ -0,0 +1,18 @@
+class Api::V1::ReviewsController < ApplicationController
```
Need a controller test for this controller

myYearOfCode replied:
We have the tests in the "user_writes_review.rspec" Should we rename that?

myYearOfCode replied:
I moved this and renamed to "spec/controllers/api/v1/reviews_controller_spec.rb.' I also added a few more response format checks. Lastly I updated all of the old "should" tests to 'expect(' tests.

CassandraNewell commented on app/controllers/stops_controller.rb:
```diff
@@ -6,7 +6,8 @@ def index
 
   def show
-    render "static_views/index"
+    @stop=Stop.find(params[:id])
```
I'm surprised by this change -- it looks like you're connecting to React through a new `show.html.erb`, which is fine, but why are you defining `@stop` and `@review` here, if you're only using `show.html.erb` to load React?

myYearOfCode replied:
You are correct. Those are not needed. They have been removed.

myYearOfCode committed:
gutting show action. 

Per Casi's comments - it only needs to render the show view (not pass params).  React is grabbing the stop id via the url.

myYearOfCode committed:
moved reviews controller to spec/controller + added a few more tests

myYearOfCode committed:
Renamed and moved the api controller tests to the right location.
